lib/icinga/apiactions.cpp

```diff
@@ -817,7 +818,7 @@ Dictionary::Ptr ApiActions::ExecuteCommand(const ConfigObject::Ptr& object,
 		execMessage->Set("method", "event::ExecuteCommand");
 		execMessage->Set("params", execParams);
 
-		listener->SyncSendMessage(endpointPtr, execMessage);
+		listener->RelayMessage(origin, checkable, execMessage, true);
```

Comment:
... using the endpoint's zone and not the checkable here?

Reply:
I just tried this 
`listener->RelayMessage(origin, endpointPtr->GetZone(), execMessage, true);`
but the message only reaches the satellite

Reply:
Does that change work with the others not undone?

Reply:
Yes

Reply:
Then please change it to the zone as the checkable may not exist on the node.